DigitalLearningSolutions.Data.Tests/DataServices/GroupsDataServiceTests.cs

```diff
@@ -446,6 +446,178 @@ public void DeleteGroup_deletes_group()
+            {
+                // Given
+                var groupId = 103;
+                var customisationId = 14675;
+                AddDelegateGroupCustomisationIfNotExist(groupId, customisationId);
+                var groupCustomisationId = (await connection.GetCustomisationsForGroup(groupId))
+                    .OrderByDescending(o => o)
+                    .First();
+
+                // When
+                groupsDataService.DeleteGroupCustomisation(groupCustomisationId);
+
+                // Then
+                var groupCustomisations = await connection.GetCustomisationsForGroup(groupId);
+                groupCustomisations.Should().NotContain(groupCustomisationId);
+            }
+            finally
+            {
+                transaction.Dispose();
+            }
+        }
+
+        [TestCase("self_enrolled", 5, 15937, 299238, 1, 0, false, 0, false)]
+        [TestCase("course_started", 5, 25918, 299238, 3, 5, false, 0, false)]
+        [TestCase("course_completed", 5, 25918, 299238, 3, 5, true, 80, true)]
+        [TestCase("non_group_candidate", 5, 25918, 298272, 3, 0, false, 0, false)]
+        public async Task RemoveRelatedProgressRecordsForCourse_does_not_update_for_case(string testCase, int groupId, int customisationId, int candidateId,
+            int enrollmentMethod, int loginCount, bool deleteStartedEnrolment, int progressDuration, bool isCompleted)
+        {
+            using var transaction = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+            try
+            {
+                // Given
+                var progressStarted = new DateTime(2021, 11, 5, 22, 23, 24, 567);
+                var progressRemoved = new DateTime(2021, 11, 6, 22, 23, 24, 567);
+                var progressCompleted = isCompleted ? new DateTime(2021, 11, 7, 22, 23, 24, 567) : (DateTime?)null;
+
+                AddDelegateGroupCustomisationIfNotExist(groupId, customisationId);
+                AddProgressIfNotExistOrIsComplete(candidateId, customisationId, progressStarted, enrollmentMethod, loginCount, progressDuration, progressCompleted);
+
+                var groupCustomisationIds = await connection.GetCustomisationsForGroupAndCustomisation(groupId, customisationId);
+
+                // When
+                groupsDataService.RemoveRelatedProgressRecordsForCourse(groupId, groupCustomisationIds.First(), deleteStartedEnrolment, progressRemoved);
+
+                // Then
+                var progressRecords = await connection.GetProgressRemovalsByCourse(customisationId);
+
+                var addedProgress = progressRecords.Where(p => p.submittedTime == progressStarted);
+                var existingProgress = progressRecords.Where(p => p.submittedTime != progressStarted);
+                addedProgress.Should().OnlyContain(p => p.removalMethodId == 0 && p.removedDate == null);
+                existingProgress.Should().OnlyContain(p => p.removedDate == null || p.removedDate != progressRemoved);

```

Comment:
I was expecting to see here specific progress records (defined by Id) that would get updated normally, but the specific reasons in each test case prevent them from being removed. So we should be able to grab specific progress records and go "this one has not been removed as expected, but this one has been removed as expected".